File: gr00t/Isaac-GR00T/examples/H20/robots/sdk_package/example_py/h10-w/motion/control_policy.py
import humanoid_sdk_py
from humanoid_sdk_py import h10w
import time

h10w_motion = h10w.H10wMotion()

print("正在进行上半身控制策略设置...")
ret_s = h10w_motion.setControlPolicy(humanoid_sdk_py.ControlPolicy.FULL_BODY)
ret_g, policy = h10w_motion.getControlPolicy()
if ret_s == 0 and ret_g == 0 and policy == humanoid_sdk_py.ControlPolicy.FULL_BODY:
    print("上半身设置为全身控制成功")
else:
    print("上半身设置为全身控制失败")

time.sleep(1)

print("正在进行上半身控制策略设置...")
ret_s = h10w_motion.setControlPolicy(humanoid_sdk_py.ControlPolicy.SINGLE_ARM)
ret_g, policy = h10w_motion.getControlPolicy()
if ret_s == 0 and ret_g == 0 and policy == humanoid_sdk_py.ControlPolicy.SINGLE_ARM:
    print("上半身设置为单臂控制成功")
else:
    print("上半身设置为单臂控制失败")

File: gr00t/Isaac-GR00T/examples/H20/robots/sdk_package/example/h10-w/motion/control_policy.cpp
// control_power.cpp

#include <iostream>
#include <iomanip>
#include <thread>
#include <chrono>
#include "h10-w/h10w_motion.h"
#include "common/common_def.h"
#include "common/LogWrapper.h"

using namespace humanoid;
using namespace humanoid::h10w;

int main() {
  std::cout << "=== 上半身设置控制策略示例 ===" << std::endl;
  H10wMotion motion;

  LOG_INFO("正在进行上半身控制策略设置...");
  humanoid::ControlPolicy policy;
  if (motion.setControlPolicy(humanoid::ControlPolicy::FULL_BODY) == 0 &&
      motion.getControlPolicy(policy) == 0 &&
      policy == humanoid::ControlPolicy::FULL_BODY) {
    LOG_INFO("上半身设置为全身控制成功");
  } else {
    LOG_ERROR("上半身设置为全身控制失败");
  }

  std::this_thread::sleep_for(std::chrono::seconds(3));

  if (motion.setControlPolicy(humanoid::ControlPolicy::SINGLE_ARM) == 0 &&
      motion.getControlPolicy(policy) == 0 &&
      policy == humanoid::ControlPolicy::SINGLE_ARM) {
    LOG_INFO("上半身设置为单臂控制成功");
  } else {
    LOG_ERROR("上半身设置为单臂控制失败");
  }

  return 0;
}